include/ignition/math/Helpers.hh

```diff
@@ -727,6 +729,42 @@ namespace ignition
       }
     }
 
+    /// \brief Convert a std::chrono::system_clock::time_point to a seconds and
+    /// nanoseconds pair.
+    /// \param[in] _time The time point to convert.
+    /// \return A pair where the first element is the number of seconds and
+    /// the second is the number of nanoseconds.
+    inline std::pair<int64_t, int64_t> timePointToSecNsec(
+        const std::chrono::system_clock::time_point &_time)
+    {
+      auto now_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+        _time.time_since_epoch());
+      auto now_s = std::chrono::duration_cast<std::chrono::seconds>(
+        _time.time_since_epoch());
+      int64_t seconds = now_s.count();
+      int64_t nanoseconds = std::chrono::duration_cast
+        <std::chrono::nanoseconds>(now_ns - now_s).count();
+      return {seconds, nanoseconds};
+    }
+
+    /// \brief Convert seconds and nanoseconds to
+    /// std::chrono::system_clock::time_point.
+    /// \param[in] _sec The seconds to convert.
+    /// \param[in] _nanosec The nanoseconds to convert.
+    /// \return A std::chrono::system_clock::time_point based on the number of
+    /// seconds and the number of nanoseconds.
+    inline std::chrono::system_clock::time_point secNsecToTimePoint(
```

Comment:
as noted in a separate comment about a failing test; I think the base for `system_clock::time_point` depends on the system, with macOS at least seeming to use microseconds rather than nanoseconds.

Reply:
relevant to @JShep1's comment: https://github.com/ignitionrobotics/ign-math/pull/150#discussion_r479505267